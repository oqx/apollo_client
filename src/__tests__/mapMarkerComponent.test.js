import React from "react";
import MapMarkerComponent from "../components/mapMarkerComponent";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import { completeShow } from "../mockResults";
import { divIcon } from "leaflet";

let store, container;

const spy = jest.fn();

const state = {
  btnClick: false
};
const handleClickStub = () => {
  state.btnClick = true;
  spy();
};

const userIcon = divIcon({
  className: "ion-android-navigate icon__navigator "
});

describe("<MapMarkerComponent />", () => {
  beforeEach(() => {
    container = shallow(
      <MapMarkerComponent
        event={completeShow}
        getPopUpStatusAndEvent={handleClickStub}
        icon={userIcon}
      />
    );
  });

  it("should render successfully", () => {
    expect(container.length).toEqual(1);
  });

  it("should render correctly", () => {
    expect(toJson(container)).toMatchSnapshot();
  });
});
