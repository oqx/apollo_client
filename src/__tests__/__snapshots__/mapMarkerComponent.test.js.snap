// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MapMarkerComponent /> should render correctly 1`] = `
<Marker
  icon={
     {
      "_initHooksCalled": true,
      "options": Object {
        "className": "ion-android-navigate icon__navigator ",
      },
    }
  }
  key="243426"
  position={
    Array [
      44.988767,
      -93.254362,
    ]
  }
>
  <Popup
    pane="popupPane"
  >
    <PopUpDispatchComponent
      event={
        Object {
          "artist": Array [
            Object {
              "displayName": "RLGDPPL",
              "id": 23431,
            },
            Object {
              "displayName": "Joelletron",
              "id": 23432,
            },
            Object {
              "displayName": "Windmill",
              "id": 23433,
            },
          ],
          "date": null,
          "distance": 4,
          "id": 243426,
          "latlng": Array [
            44.988767,
            -93.254362,
          ],
          "radius": 10,
          "type": "Concert",
        }
      }
      getPopUpStatusAndEvent={[Function]}
    />
  </Popup>
</Marker>
`;
